src/pages/App.jsx

```diff
@@ -1,28 +1,62 @@
+import { useEffect, useState } from "react";
 import { Route, Routes } from "react-router-dom";
 
 import Header from "../components/Layout/Header";
+import Loading from "../components/UI/Loading";
 import ReactQueryProviders from "../config/ReactQueryProvider";
+import useBoundStore from "../store/client/useBoundStore";
 import GroupPage from "./GroupPage";
 import HomePage from "./HomePage";
 import KeywordPage from "./KeywordPage";
 import MyPage from "./MyPage";
 import NotFoundPage from "./NotFoundPage";
+import { getAuth, onAuthStateChanged } from "firebase/auth";
 
 const App = () => {
+  const setIsSignIn = useBoundStore((state) => state.setIsSignIn);
+  const setUserInfo = useBoundStore((state) => state.setUserInfo);
+  const signOut = useBoundStore((state) => state.signOut);
+  const [isAuthChecked, setIsAuthChecked] = useState(false);
+
+  useEffect(() => {
+    const auth = getAuth();
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      if (user) {
+        const uid = user.uid;
+        const { email, displayName, photoUrl: photoURL } = user.reloadUserInfo;
```

Comment:
reloadUserInfo는 localStorage에서의 사용자 정보를 가져오는건가요?

Reply:
앗, 제가 추가 코멘트를 달아두었다면 더 좋았겠네요! 코멘트 감사드립니다 :) cc. @only-pine 

정확히는 `user`로 localStorage에 저장된 사용자 정보에 접근해요. `onAuthStateChanged` 두 번째 인자인 콜백 함수의 매개변수로 받는 `user`에 토큰 비롯한 로그인 했던 사용자 정보들이 담겨있어요. (아래 이미지 첨부) 아래서 더 설명 드리겠지만, 결론적으로는 `reloadUserInfo` 프로퍼티 접근으로 구현 후 추가로 검증된 근거들이 확인되면 업데이트 해보면 좋겠다는 생각이에요.

`reloadUserInfo` 프로퍼티에 접근하는 방향, `user` 로만 접근하는 방향 모두 가능해요. 두 가지 모두 레퍼런스들을 참고할 수 있었어요. 후자인 경우에도 아래처럼 충분히 가능하고요.

```suggestion
        const { uid, email, displayName, photoURL } = user;
```

다만 [Firebase API 공식 문서](https://firebase.google.com/docs/reference/js/auth.auth#methods)나 리서치 통해서 2개의 차이를 뚜렷하게 발견하지 못 했어요. 두 가지 모두 유효하다고 보아서 결국 `reloadUserInfo`프로퍼터 접근하기로 결정했어요. 맥락은 (아래 이미지를 참고해주세요) `reloadUserInfo` 프로퍼티가 `lastSignInAt`나 `localId` 등 이전 로그인 이력 및 환경 정보를 유일하게 담고 있기 때문이에요! 그래서 해당 프로퍼터 접근하는 것이, `onAuthStateChanged` 옵저버를 통해 이전 이력을 업데이트 하고자 하는 의도에 맞다고 봤어요 🙂 

의견 있으시거나 이해가 어려운 부분은 추가로 말씀 부탁 드려요! 함께 검토해보면 좋겠습니다 ㅎㅎ

[`onAuthStateChanged` 두 번째 인자인 콜백 함수의 매개변수로 받는 `user`] 
<image width="400" src="https://github.com/user-attachments/assets/fe42c10d-1ab3-40da-9823-a9fa6ade2e7a">

Reply:
@jin-ttao 답변 감사합니다! NextOrObserver 콜백의 파라미터인 user 객체에 대해서 서버 쪽 인증 검증을 한 번 거치는 로직이 추가되어도 좋겠네요. :)